trio/_socket.py

```diff
@@ -575,7 +719,19 @@ async def _resolve_address_nocp(self, address, *, local):
             local=local,
         )
 
-    async def _nonblocking_helper(self, fn, args, kwargs, wait_fn):
+    # args and kwargs must be starred, otherwise pyright complains:
+    # '"args" member of ParamSpec is valid only when used with *args parameter'
+    # '"kwargs" member of ParamSpec is valid only when used with **kwargs parameter'
+    # wait_fn and fn must also be first in the signature
+    # 'Keyword parameter cannot appear in signature after ParamSpec args parameter'
+
```

Comment:
FWIW this isn't just a pyright thing as both these requirements come directly from the Parameter Specification PEP.

Reply:
Huh, weird that mypy didn't complain about it then.